test/precipitation_susceptibility_tests.jl

```diff
@@ -0,0 +1,115 @@
+import Test as TT
+
+import CloudMicrophysics as CM
+import CloudMicrophysics.Parameters as CMP
+import CloudMicrophysics.CommonTypes as CMT
+import CloudMicrophysics.Microphysics0M as CM0
+import CloudMicrophysics.Microphysics1M as CM1
+import CloudMicrophysics.Microphysics2M as CM2
+import CLIMAParameters as CP
+using ForwardDiff
+
+include(joinpath(pkgdir(CM), "test", "create_parameters.jl"))
+
+const APS = CMP.AbstractCloudMicrophysicsParameters
+
+const FT = Float64
+
+"""
+A structure containing the logarithmic derivatives of the production of
+precipitation with respect to the specific humidities and number
+densities of liquid and rain water.
+"""
+Base.@kwdef struct precip_susceptibility_rates{FT<:Real}
+    d_ln_pp_d_ln_q_liq::FT
+    d_ln_pp_d_ln_q_rai::FT
+    d_ln_pp_d_ln_N_liq::FT
+    d_ln_pp_d_ln_N_rai::FT
+end
+
+"""
+Returns the precipitation susceptibility rates due to autoconversion,
+using automatic differentiation.
+Works for any 2-moment scheme, as long as autoconversion is defined for it.
+"""
+function precipitation_susceptibility_autoconversion(
+    param_set::APS,
```

Comment:
Would it be possible to define it such that you pass the function as one of the arguments?